nms/app/packages/magmalte/app/components/context/RefreshContext.js

```
@@ -0,0 +1,305 @@
+  if (type === 'subscriber') {
+    return fetchSubscribers({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+    });
+  } else if (type === 'gateway') {
+    return fetchGateways({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+      enqueueSnackbar: enqueueSnackbar,
+    });
+  } else {
+    return fetchEnodebs({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+      enqueueSnackbar: enqueueSnackbar,
+    });
+  }
+}
+
+async function fetchEnodebs(props: fetchProps) {
+  const {networkId, refreshingState, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== '') {
+    let newEnb = {};
+    try {
+      newEnb = await MagmaV1API.getLteByNetworkIdEnodebsByEnodebSerial({
+        networkId: networkId,
+        enodebSerial: id,
+      });
+
+      if (Object.keys(newEnb).length) {
+        const newEnbSt = await MagmaV1API.getLteByNetworkIdEnodebsByEnodebSerialState(
+          {
+            networkId: networkId,
+            enodebSerial: id,
+          },
+        );
+        return {
+          ...refreshingState,
+          state: {
+            ...refreshingState.state,
+            enbInfo: {
+              ...refreshingState.state.enbInfo,
+              [id]: {enb_state: newEnbSt ?? {}, enb: newEnb},
+            },
+          },
+        };
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb informations', {
+        variant: 'error',
+      });
+      return {
+        ...refreshingState,
+        state: {
+          ...refreshingState.state,
+          enbInfo: {
+            ...refreshingState.state.enbInfo,
+            [id]: {enb_state: {}, enb: newEnb},
+          },
+        },
+      };
+    }
+  } else {
+    let enb = {};
+    try {
+      enb = await MagmaV1API.getLteByNetworkIdEnodebs({networkId});
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb information', {
```

let's ignore this error